amqp/src/test/java/akka/stream/alpakka/amqp/javadsl/AmqpConnectorsTest.java

```diff
@@ -248,128 +245,163 @@ public void publishAndConsumeRpcWithoutAutoAck() throws Exception {
     final String queueName = "amqp-conn-it-spec-rpc-queue-" + System.currentTimeMillis();
     final QueueDeclaration queueDeclaration = QueueDeclaration.create(queueName);
 
-    final Flow<OutgoingMessage,IncomingMessage, CompletionStage<String>> ampqRpcFlow = AmqpRpcFlow.atMostOnceFlow(
-            AmqpSinkSettings.create().withRoutingKey(queueName).withDeclarations(queueDeclaration), 10);
+    final List<String> input = Arrays.asList("one", "two", "three", "four", "five");
 
-    final Integer bufferSize = 10;
-    final Source<CommittableIncomingMessage, NotUsed> amqpSource = AmqpSource.committableSource(
-            NamedQueueSourceSettings.create(
-                    DefaultAmqpConnection.getInstance(),
-                    queueName
-            ).withDeclarations(queueDeclaration),
-            bufferSize
-    );
+    final Flow<OutgoingMessage, CommittableIncomingMessage, CompletionStage<String>> ampqRpcFlow = AmqpRpcFlow.committableFlow(
+        AmqpSinkSettings.create()
+            .withRoutingKey(queueName)
+            .withDeclarations(queueDeclaration)
+        , 10, 1);
+    Pair<CompletionStage<String>, TestSubscriber.Probe<IncomingMessage>> result =
+      Source.from(input)
+        .map(ByteString::fromString)
+        .map(bytes -> new OutgoingMessage(bytes, false, false, Optional.empty(), Optional.empty()))
+        .viaMat(ampqRpcFlow, Keep.right())
+        .mapAsync(1, cm -> cm.ack(false).thenApply(unused -> cm.message()))
+        .toMat(TestSink.probe(system), Keep.both())
+        .run(materializer);
 
-    final List<String> input = Arrays.asList("one", "two", "three", "four", "five");
-    TestSubscriber.Probe<IncomingMessage> probe =
-            Source.from(input)
-                    .map(ByteString::fromString)
-                    .map(bytes -> new OutgoingMessage(bytes, false, false, Optional.empty(), Optional.empty()))
-                    .via(ampqRpcFlow)
-                    .runWith(TestSink.probe(system), materializer);
+    result.first().toCompletableFuture().get(5, TimeUnit.SECONDS);
 
-    Sink<OutgoingMessage, CompletionStage<Done>> amqpSink = AmqpSink.createReplyTo(
-            AmqpReplyToSinkSettings.create(DefaultAmqpConnection.getInstance())
+    Sink<OutgoingMessage, CompletionStage<Done>> amqpSink = AmqpSink.createReplyTo(AmqpReplyToSinkSettings.create(DefaultAmqpConnection.getInstance()));
+
+    final Source<IncomingMessage, NotUsed> amqpSource = AmqpSource.atMostOnceSource(
```

Comment:
The actual issue wasn't on the stage but on the Java test, I think.

This line was a `committableSource` instead of an `atMostOnceSource` leaving unacked messages in the queue and the buffer size was 10 instead of 1 which might be making the stage pulling one element too many from the source and causing the timeout.
These were the only differences I could find between the scala test and the failing java test.